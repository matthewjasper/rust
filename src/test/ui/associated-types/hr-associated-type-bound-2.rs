trait X<'a>
where
    for<'b> <Self as X<'b>>::U: Clone,
{
    type U: ?Sized;
    fn f(&self, x: &Self::U) {
        <Self::U>::clone(x);
    }
}

impl X<'_> for u32
//~^ ERROR the trait bound `str: std::clone::Clone` is not satisfied
where
    for<'b> <Self as X<'b>>::U: Clone,
{
    type U = str;
    //~^ ERROR the trait bound `str: std::clone::Clone` is not satisfied
}

fn main() {
    1u32.f("abc");
}
