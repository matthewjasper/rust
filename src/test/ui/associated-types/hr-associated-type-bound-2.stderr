error[E0277]: the trait bound `str: std::clone::Clone` is not satisfied
  --> $DIR/hr-associated-type-bound-2.rs:16:14
   |
LL | trait X<'a>
   |       - required by a bound in this
LL | where
LL |     for<'b> <Self as X<'b>>::U: Clone,
   |                                 ----- required by this bound in `X`
...
LL |     type U = str;
   |              ^^^ the trait `std::clone::Clone` is not implemented for `str`

error[E0277]: the trait bound `str: std::clone::Clone` is not satisfied
  --> $DIR/hr-associated-type-bound-2.rs:11:6
   |
LL | / trait X<'a>
LL | | where
LL | |     for<'b> <Self as X<'b>>::U: Clone,
LL | | {
...  |
LL | |     }
LL | | }
   | |_- required by this bound in `X`
LL | 
LL |   impl X<'_> for u32
   |        ^^^^^ the trait `std::clone::Clone` is not implemented for `str`
   |
   = note: required because of the requirements on the impl of `X<'_>` for `u32`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
